lib/lf/uscalfe/fe_space_hp.h

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file
+ * @brief Lagrangian finite elements of arbitrary polynomials degree
+ * @author Tobias Rohner
+ * @date   May 20202
+ * @copyright MIT License
+ */
+
+#ifndef LF_USCALFE_FE_SPACE_HP_H_
+#define LF_USCALFE_FE_SPACE_HP_H_
+
+#include <lf/mesh/utils/all_codim_mesh_data_set.h>
+#include "hp_fe.h"
+#include "scalar_fe_space.h"
+
+namespace lf::uscalfe {
+
+/**
+ * @brief Lagrangian Finite Element Space of arbitrary degree
+ */
+template <typename SCALAR>
+class FeSpaceHP : public ScalarFESpace<SCALAR> {
+ public:
+  using Scalar = SCALAR;
+
+  FeSpaceHP() = delete;
+  FeSpaceHP(const FeSpaceHP &) = delete;
+  FeSpaceHP(FeSpaceHP &&) noexcept = default;
+  FeSpaceHP &operator=(const FeSpaceHP &) = delete;
+  FeSpaceHP &operator=(FeSpaceHP &&) noexcept = default;
+
+  /**
+   * @brief Constructor: Sets up the dof handler
+   * @param mesh_p A shared pointer to the underlying mesh (immutable)
+   * @param N The polynomial degree of the Finite Element Space
+   */
+  explicit FeSpaceHP(std::shared_ptr<const lf::mesh::Mesh> mesh_p, unsigned N)
```

Comment:
Here clang-tidy complains: 
~~~
error: the parameter 'mesh_p' is copied for each invocation but only used as a const reference; consider making it a const reference [performance-unnecessary-value-param,-warnings-as-errors]
  explicit FeSpaceHP(std::shared_ptr<const lf::mesh::Mesh> mesh_p, unsigned N)
                                                           ^
                     const                                &
~~~

I have no clue why?

Reply:
This is a mistake of clang-tidy: It somehow assumes that `std::shared_ptr` is expensive to move which is not the case. I've fixed this by editing `.clang-tidy` in https://github.com/craffael/lehrfempp/commit/23ca2e5109ac85c7c9ec88abadaf56412caddced